src/main/java/org/jenkinsci/plugins/mesos/MesosCloud.java

```diff
@@ -65,13 +88,36 @@ public MesosCloud(String name) throws InterruptedException, ExecutionException {
    *
    * @return A future reference to the launched node.
    */
-  private CompletableFuture<Node> startAgent() {
-    // mesos.enqueueAgent().thenCompose(mesosSlave -> mesosSlave.waitUntilOnlineAsync());
-    throw new NotImplementedException();
-  }
-
   @Override
   public boolean canProvision(Label label) {
-    throw new NotImplementedException();
+    // TODO: implement executor limits
+    return true;
+  }
+
+  public MesosApi getMesosClient() {
+    return this.mesos;
+  }
+
+  /**
+   * Start a Jenkins agent.jar on Mesos.
+   *
+   * <p>Provide a callback for Jenkins to start a Node.
+   *
+   * @return A future reference to the launched node.
+   */
+  public Future<Node> startAgent() throws Exception {
+    return mesos
+        .enqueueAgent(this, 0.1, 32)
+        .thenCompose(
+            mesosSlave -> {
+              try {
+                Jenkins.getInstanceOrNull().addNode(mesosSlave);
+                logger.info("waiting for slave to come online...");
+              } catch (Exception e) {
+                logger.info("error occured when waiting for slave to come online...");
```

Comment:
We should log the exception and decide on a crash strategy. I think we should just not catch the error for now.

Reply:
changed to only handle IOexception due to the Jenkins.getInstance() call, not sure how to handle exception from inside the lambda and was having issues with that